firebase/best-practices.md

```diff
@@ -61,3 +61,36 @@ exports.getUser = functions.https.onRequest((req, res) => {
 
 // => /getUser?uid=npWxk05ZCKMcYb0OaDSJffYQZZq1
 ```
+
+#### Let the `Promise` fail to construct meaningful logs in Firebase unless there is a fallback strategy
+
+GOOD
+
+```js
+exports.signup = functions.auth.user().onCreate(user =>
+  Model.create(user)
+    .then(() => Service.createCustomer(user))
+    .catch(err => {
+      console.error(JSON.stringify(err));
+
+      // Re-throw the error to fail the Promise explicitly.
+      throw err;
+    })
+);
+
+// ERROR => Function execution took 60 ms, finished with status: 'error'
+```
+
+BAD
+
+```js
+exports.signup = functions.auth.user().onCreate(user =>
+  Model.create(user)
+    .then(() => Service.createCustomer(user))
+    .catch(err => {
+      console.error(JSON.stringify(err));
+    })
+);
+
+// ERROR => Function execution took 60 ms, finished with status: 'ok'
```

Comment:
I think this doesn't say `ERROR` right?

Reply:
This `ERROR` does not represent the actual error message/object, I have added this extra stuff for some reason. But agreed, in the second case it is unnecessary, perhaps we can even remove it from the other case as well as it can be misleading.

Reply:
Then I'd propose to just use `// => Function execution took 60 ms, finished with status: 'ok'` for both cases.

Reply:
The first case should be `// => Function execution took 60 ms, finished with status: 'error'`.

Reply:
:+1: